work/test/load/cli/cli.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"slices"
+	"time"
+
+	"github.com/tidepool-org/platform/pointer"
+	"github.com/tidepool-org/platform/work"
+	"github.com/tidepool-org/platform/work/load"
+	"github.com/urfave/cli/v3"
+)
+
+func main() {
+
+	var filePath string
+	var outputDir string
+	var urlBase string
+
+	var generateDurationSeconds int64
+	var generateCount int64
+	var generateResult string
+
+	var baseURLFlag = &cli.StringFlag{
+		Name:        "urlBase",
+		Aliases:     []string{"u"},
+		Value:       "https://qa2.development.tidepool.org",
+		Usage:       "base URL for environment we are testing against",
+		Destination: &urlBase,
+		Required:    true,
+	}
+
+	var outputDirFlag = &cli.StringFlag{
+		Name:        "outputDir",
+		Aliases:     []string{"o"},
+		Usage:       "directory to save the test output",
+		Destination: &outputDir,
+		Action: func(ctx context.Context, cmd *cli.Command, v string) error {
+			if _, err := os.Stat(v); err != nil {
+				if os.IsNotExist(err) {
+					os.Mkdir(v, 0755)
+				}
+				return nil
+			}
+			return nil
+		},
+	}
+
+	cmd := &cli.Command{
+		Usage: "Load test the work system",
+		Commands: []*cli.Command{
+			{
+				Name:    "run",
+				Aliases: []string{"r"},
+				Usage:   "run the load test",
+				Flags: []cli.Flag{
+					baseURLFlag,
+					&cli.StringFlag{
+						Name:    "filePath",
+						Aliases: []string{"f"},
+						Usage:   "path to the load test file",
+						Action: func(ctx context.Context, cmd *cli.Command, v string) error {
+							if _, err := os.Stat(v); err != nil {
+								return fmt.Errorf("filePath %s does not exist", v)
+							}
+							return nil
+						},
+						Destination: &filePath,
+					},
+					outputDirFlag,
+				},
+				Action: func(ctx context.Context, cmd *cli.Command) error {
+
+					testData, err := os.ReadFile(filePath)
+					if err != nil {
+						return fmt.Errorf("error loading %s %s", filePath, err.Error())
+					}
+
+					url := fmt.Sprintf("%s/v1/work/load", urlBase)
+
+					req, err := http.NewRequest("POST", url, bytes.NewReader(testData))
+					if err != nil {
+						return fmt.Errorf("error sending data to load work %s", err.Error())
+					}
+					req.Header.Add("content-type", "application/json")
+					req.Header.Add("accept", "application/json")
+
+					res, err := http.DefaultClient.Do(req)
+					if err != nil {
+						return fmt.Errorf("unable to issue work load test API request: %s", err.Error())
```

Comment:
For all of the code where you use `fmt.Errorf("... %s", err.Error())` consider instead `fmt.Errorf("... %w", err)`.

Reply:
Note using `%w` for the `err`.